addons/web/static/src/js/views/list/list_controller.js

```diff
@@ -203,6 +240,54 @@ var ListController = BasicController.extend({
             this._updatePaging(state);
         }
     },
+    /**
+     * @private
+     * @param {jQueryElement} $el
+     * @param {Object} node
+     */
+    _addOnHeaderButtonClickAction: function ($el, node) {
+        if (node.attrs.special || node.attrs.confirm || node.attrs.type) {
+            $el.on("click", async () => {
+                const record = this.model.get(this.handle);
+                const selectedResIDs = this.getSelectedIds();
+                this._disableButtons();
+
+                // In a nutshell: append the context of the button
+                // and our custom one (active_ids and domain) to the action's execution context
+                const actionData = Object.assign({}, node.attrs, {
+                    context: record.getContext({ additionalContext: node.attrs.context || {} })
+                });
+                // The action's execution context must contain
+                // our custom context keys (active_ids, domain etc...)
+                Object.assign(actionData.context, {
+                    active_domain: record.getDomain(),
+                    active_id: selectedResIDs[0],
+                    active_ids: selectedResIDs,
+                    active_model: record.model,
+                });
+                // In a nutshell: load the action with the correct context
+                // and record parameters (resIDs, model etc...)
+                const recordData = {
+                    context: record.getContext(),
+                    model: record.model,
+                    resIDs: selectedResIDs,
+                };
+                try {
+                    if (this.isDomainSelected) {
+                        const domainResIDs = await this._domainToResIds(record.getDomain(), session.active_ids_limit);
+                        recordData.resIDs = domainResIDs;
+                        Object.assign(actionData.context, {
+                            active_id: domainResIDs[0],
+                            active_ids: domainResIDs,
```

Comment:
do this earlier to build (or at least override) actionData.context only once?

Reply:
yes, i'd also prefer that, but I think I prefer the try/finally below more to ensure buttons are re-enabled, and put that code only in one place...